src/jitlog/generator.lua

```diff
@@ -137,13 +137,45 @@ local function parse_field(field)
   return name, typename, nil, length
 end
 
+local function parse_structcopy(msgdef, structcopy, fieldlookup)
+  local arg_name, arg_type = parse_field(structcopy.arg)
+  local struct_arg = arg_type..arg_name
+
+  if structcopy.store_address then
+    local struct_addr = fieldlookup[structcopy.store_address]
+    assert(struct_addr, "store_address field '" .. structcopy.store_address .. "' does not exist")
+    struct_addr.noarg = true
+    struct_addr.struct_addr = true
+    struct_addr.value_name = arg_name
+  end
+
+  for name, struct_field in pairs(structcopy.fields) do
+    if type(name) == "number" then
+      -- Array entrys mean we copy the value in to a field with the same name as the struct field
```

Comment:
Not sure I understand the comment.

Reply:
its a mixed hash table and array. The array entries will have a key thats a number

Reply:
OK, can we clarify  the comment?